java/client/src/main/java/glide/managers/CommandManager.java

```diff
@@ -392,4 +385,43 @@ private Response exceptionHandler(Throwable e) {
         }
         throw new RuntimeException(e);
     }
+
+    /**
+     * Add the given set of arguments to the output Command.Builder.
+     *
+     * @param arguments         The arguments to add to the builder.
+     * @param outputBuilder     The builder to populate with arguments.
+     */
+    private static void populateRequestWithArgs(String[] arguments, Command.Builder outputBuilder) {
+        populateRequestWithArgs(Arrays.stream(arguments)
+            .map(value -> value.getBytes(StandardCharsets.UTF_8)).toList(), outputBuilder);
+    }
+
+    /**
+     * Add the given set of arguments to the output Command.Builder.
+     *
+     * @param arguments         The arguments to add to the builder.
+     * @param outputBuilder     The builder to populate with arguments.
+     */
+    private static void populateRequestWithArgs(GlideString[] arguments, Command.Builder outputBuilder) {
+        populateRequestWithArgs(Arrays.stream(arguments).map(GlideString::getBytes).toList(), outputBuilder);
+    }
+
+    /**
+     * Add the given set of arguments to the output Command.Builder.
+     *
+     * @param arguments         The arguments to add to the builder.
+     * @param outputBuilder     The builder to populate with arguments.
+     */
+    private static void populateRequestWithArgs(List<byte[]> arguments, Command.Builder outputBuilder) {
+        final long totalArgSize = arguments.stream().mapToLong(arg -> arg.length).sum();
+        if (totalArgSize < RedisValueResolver.MAX_REQUEST_ARGS_LENGTH) {
```

Comment:
Should we count total args amount in a transaction?
Let say `MAX_REQUEST_ARGS_LENGTH` is 100 and transaction has 20 commands with 20 args each. In that case request will be shipped with 20 arrays of args instead of 20 pointers.

Reply:
Does the concept of a transaction exist for Python? The Python PR didn't have this functionality. I'm inclined to treat that as a separate-but-related feature.

Reply:
No idea unfortunately

Reply:
Transactions do exist for Python. I created #1713 to track this.

Reply:
BTW, to be clear, it's not the total number of arguments in a command that is used to determine if a pointer is used. It's the total size in bytes of the arguments (that's why in the test case, we have two arguments to set() but they are 33MB each).